packages/contracts-core/contracts/manager/AgentManager.sol

```diff
@@ -65,6 +64,13 @@ abstract contract AgentManager is MessagingBase, AgentManagerEvents, IAgentManag
         _;
     }
 
+    modifier onlyWhenStuck() {
+        // Check if there has been no fresh data from the Notaries for a while.
+        (uint40 snapRootTime,,) = InterfaceDestination(destination).destStatus();
+        if (block.timestamp < FRESH_DATA_TIMEOUT + snapRootTime) revert NotStuck();
```

Comment:
`snapRootTime` from destination is cast to a `uint40`. if `block.timestamp > type(uint40).max`, the only when stuck check won't be valid (always returns `true` as stuck).

should cast `block.timestamp` here to `uint40(block.timestamp)` to be consistent and avoid this edge case.

Reply:
I would argue that it's the responsibility of the contract that is producing data (Destination) rather than the one that is consuming it (AgentManager). 

The #1457 fixes the safe casts so that it's not possible to save the corrupted data, and the amount of bits to allocate to timestamps is tracked in #1429